st2common/st2common/services/workflows.py

```diff
@@ -535,17 +535,24 @@ def request_action_execution(wf_ex_db, task_ex_db, st2_ctx, ac_ex_req, delay=Non
     # Identify the runner for the action.
     runner_type_db = action_utils.get_runnertype_by_name(action_db.runner_type['name'])
 
+    # Identify action pack name
+    action_ref = task_ex_db.task_spec.get('spec').get('action')
+    if action_ref:
+        pack = action_ref.split('.')[0]
+    else:
+        pack = st2_ctx.get('pack')
+
     # Set context for the action execution.
     ac_ex_ctx = {
-        'pack': st2_ctx.get('pack'),
+        'pack': pack,
         'user': st2_ctx.get('user'),
         'parent': st2_ctx,
         'orquesta': {
             'workflow_execution_id': str(wf_ex_db.id),
             'task_execution_id': str(task_ex_db.id),
             'task_name': task_ex_db.task_name,
             'task_id': task_ex_db.task_id
-        }
+        },
```

Comment:
Why do we need a comma here?

Reply:
Did run `nosetests --nocapture st2common/tests/unit/test_param_utils.py`  and all passed